src/naive_bayes/baseNaiveBayes.ts

```diff
@@ -0,0 +1,149 @@
+/**
+*  @license
+* Copyright 2022, JsData. All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+* ==========================================================================
+*/
+import { polyfillUnique } from '../tfUtils'
+import { tf } from '../shared/globals'
+import { Scikit1D, Scikit2D } from '../types'
+import { convertToNumericTensor2D, convertToTensor1D } from '../utils'
+import { Tensor1D } from '@tensorflow/tfjs-core'
+
+export interface NaiveBayesParams {
+  /**
+   * Prior probabilities of the classes. If specified the priors are not adjusted according to the data.
+   */
+  priors?: Scikit1D
+
+  /**
+   * Portion of the largest variance of all features that is added to variances for calculation stability.
+   * Default value is 1e-9
+   */
+  varSmoothing?: number
+}
+
+export abstract class BaseNaiveBayes {
+  priors?: tf.Tensor1D
+  varSmoothing: number
+
+  private classes: tf.Tensor1D
+  private means: tf.Tensor1D[]
+  private variances: tf.Tensor1D[]
+
+  constructor(params: NaiveBayesParams) {
+    this.classes = tf.tensor1d([])
+    this.means = []
+    this.variances = []
+    if (params.priors) {
+      this.priors = convertToTensor1D(params.priors)
+    }
+    this.varSmoothing = params.varSmoothing ? params.varSmoothing : 1e-9
+  }
+
+  /**
+   * Train the model by calculating the mean and variance of sample distribution.
+   * @param X
+   * @param y
+   * @returns
+   */
+  public async fit(X: Scikit2D, y: Scikit1D) {
+    const features = convertToNumericTensor2D(X)
+    const labels = convertToTensor1D(y)
+
+    const {values, meansByLabel, variancesByLabel} = tf.tidy(() => {
+      polyfillUnique(tf)
+      const meansByLabel: Tensor1D[] = []
+      const variancesByLabel: Tensor1D[] = []
+
+      // Get the list of unique labels
+      const {values, indices} = tf.unique(labels)
+
+      const {mean, variance} = tf.moments(features, 0)
```

Comment:
`const { variance } = tf.moments(features, 0)`
We don't use mean either. My vote is to just get rid of them, and save the computer some work.